src/lib/twizzler-driver/src/dma.rs

```diff
@@ -0,0 +1,284 @@
+use std::{
+    alloc::Layout,
+    collections::BTreeMap,
+    sync::{Arc, Mutex},
+};
+
+use twizzler_abi::{
+    kso::{KactionCmd, KactionFlags, KactionGenericCmd},
+    marker::BaseType,
+    object::{MAX_SIZE, NULLPAGE_SIZE},
+    syscall::{sys_kaction, BackingType, LifetimeType},
+};
+use twizzler_object::{CreateSpec, Object};
+
+struct DmaBase;
+
+impl BaseType for DmaBase {
+    fn init<T>(_t: T) -> Self {
+        Self
+    }
+    fn tags() -> &'static [(
+        twizzler_abi::marker::BaseVersion,
+        twizzler_abi::marker::BaseTag,
+    )] {
+        todo!()
+    }
+}
+
+struct DmaObjectInner {
+    object: Object<DmaBase>,
+    huge_regions: Vec<usize>,
+    page_regions: Vec<usize>,
+    small_offsets: Vec<(usize, usize)>,
+    map_huge: BTreeMap<usize, u64>,
+    map_page: BTreeMap<usize, u64>,
+}
+
+// TODO
```

Comment:
I guess you could say what is to be done.

Reply:
:) yes, you're right. I intend to write up a separate RFC for this part.